### PR TITLE
[srp-server] add more error tracing logs

### PR DESCRIPTION
**Pull request by wgtdkp**
This commit adds more error tracing logs to help identify where a
SRP service registration fails. The log format is also updated to
capitalize the first char to align with rest of the codebase.

### CODE REVIEW

File: src/core/net/srp_server.cpp [hunk 1]
```diff
@@ -344,13 +349,13 @@ void Server::HandleServiceUpdateResult(ServiceUpdateId aId, Error aError)
...
     {
-        LogInfo("delayed SRP host update result, the SRP update has been committed (updateId = %u)", aId);
+        LogInfo("Delayed SRP host update result, the SRP update has been committed (updateId = %u)", aId);
     }
 }
 
 void Server::HandleServiceUpdateResult(UpdateMetadata *aUpdate, Error aError)
 {
-    LogInfo("handler result of SRP update (id = %u) is received: %s", aUpdate->GetId(), otThreadErrorToString(aError));
+    LogInfo("Handler result of SRP update (id = %u) is received: %s", aUpdate->GetId(), otThreadErrorToString(aError));
```
Comment (abtink) [inline src/core/net/srp_server.cpp:358]: Can use `ErrorToString()` instead?

File: src/core/net/srp_server.cpp [hunk 2]
```diff
@@ -434,17 +439,15 @@ void Server::CommitSrpUpdate(Error                    aError,
     }
     else
     {
-        LogInfo("add new host %s", aHost.GetFullName());
+        AddHost(aHost);
+        shouldFreeHost = false;
```
Comment (abtink) [inline src/core/net/srp_server.cpp:443]: Just double that changing the order and calling `AddHost()` before `for` loop is perfectly fine (adding a note to help other reviewer).

I One thing we can consider (I suggest in a future PR): I think `AddHost()` is only called from here and just logs and adds the host entry in the linked list so we can actually just do this directly here and remove the method

File: src/core/net/srp_server.cpp [hunk 3]
```diff
@@ -1243,12 +1281,12 @@ void Server::SendResponse(const Dns::UpdateHeader &aHeader,
...
     SuccessOrExit(error = GetSocket().SendTo(*response, aMessageInfo));
 
-    LogInfo("send response with granted lease: %u and key lease: %u", aLease, aKeyLease);
+    LogInfo("Send success response with granted lease: %u and key lease: %u", aLease, aKeyLease);
 
 exit:
     if (error != kErrorNone)
     {
-        LogWarn("failed to send response: %s", ErrorToString(error));
+        LogCrit("Failed to send response: %s", otThreadErrorToString(error));
```
Comment (abtink) [inline src/core/net/srp_server.cpp:1289]: Use `ErrorToString()`?

**Review by abtink — APPROVED** [review]
Looks great! Thanks @wgtdkp 👍 

Couple of smaller suggestions below.

File: src/core/net/srp_server.cpp (hunk 2 above)
Comment (wgtdkp) [reply src/core/net/srp_server.cpp:443]: Thanks for confirming the order!

I think it also makes sense to keep the `AddHost` method since there is a symmetrical `RemoveHost` method :)

File: src/core/net/srp_server.cpp [hunk 4]
```diff
@@ -722,6 +725,11 @@ Error Server::ProcessZoneSection(const Message &aMessage, MessageMetadata &aMeta
     aMetadata.mOffset = offset;
 
 exit:
+    if (error != kErrorNone)
+    {
+        LogCrit("Failed to process DNS Zone section: %s", otThreadErrorToString(error));
```
[the remark this answers is not in the document]
Comment (wgtdkp) [reply src/core/net/srp_server.cpp]: Thanks, updated!

File: src/core/net/srp_server.cpp [hunk 5]
```diff
@@ -323,9 +326,11 @@ bool Server::HasNameConflictsWith(Host &aHost) const
 
         for (const Host &host : mHosts)
         {
-            if (host.HasServiceInstance(service.GetInstanceName()))
+            if (host.HasServiceInstance(service.GetInstanceName()) &&
```
Comment (wgtdkp) [inline src/core/net/srp_server.cpp:329]: @abtink I think you change here to match against instance name. But shouldn't the full name be used here? For example, `abcdef._meshcop._udp.local` should not conflict with `abcdef._ipps._tcp.local`.

The SRP draft 12 statement about this is:
> [2.3.3. ](https://www.ietf.org/archive/id/draft-ietf-dnssd-srp-12.html#section-2.3.3)[FCFS Name And Signature Validation](https://www.ietf.org/archive/id/draft-ietf-dnssd-srp-12.html#name-fcfs-name-and-signature-val)
Assuming that a DNS Update message has been validated with these conditions and is a valid SRP Update, the server checks that the name in the Host Description Instruction exists. If so, then the server checks to see if the KEY record on that name is the same as the KEY record in the Host Description Instruction. The server performs the same check for the KEY records in any Service Description Instructions.

I think name in the service description instruction is the full name but not the instance name. Thoughts?

Comment (simonlingoogle) [reply src/core/net/srp_server.cpp:329]: I think it should be the full name.

It should be fine to have both `Floor 1._printer._tcp` and `Floor 1._light._tcp`.

File: src/core/net/srp_server.cpp [hunk 6]
```diff
@@ -722,6 +725,11 @@ Error Server::ProcessZoneSection(const Message &aMessage, MessageMetadata &aMeta
     aMetadata.mOffset = offset;
 
 exit:
+    if (error != kErrorNone)
+    {
+        LogCrit("Failed to process DNS Zone section: %s", ErrorToString(error));
```
Comment (simonlingoogle) [inline src/core/net/srp_server.cpp:730]:
```suggestion
        LogWarn("Failed to process DNS Zone section: %s", ErrorToString(error));
```
Failing to process a DNS message should not be critical for the whole OT stack?

File: src/core/net/srp_server.cpp [hunk 7]
```diff
@@ -747,6 +755,11 @@ Error Server::ProcessUpdateSection(Host &aHost, const Message &aMessage, Message
     VerifyOrExit(!HasNameConflictsWith(aHost), error = kErrorDuplicated);
 
 exit:
+    if (error != kErrorNone)
+    {
+        LogCrit("Failed to process DNS Update section: %s", ErrorToString(error));
```
Comment (simonlingoogle) [inline src/core/net/srp_server.cpp:760]: ditto

File: src/core/net/srp_server.cpp [hunk 8]
```diff
@@ -823,6 +836,11 @@ Error Server::ProcessHostDescriptionInstruction(Host &                 aHost,
     // the host is being removed or registered.
 
 exit:
+    if (error != kErrorNone)
+    {
+        LogCrit("Failed to process Host Description instructions: %s", ErrorToString(error));
```
Comment (simonlingoogle) [inline src/core/net/srp_server.cpp:841]: ditto

File: src/core/net/srp_server.cpp [hunk 9]
```diff
@@ -893,6 +911,11 @@ Error Server::ProcessServiceDiscoveryInstructions(Host &                 aHost,
     }
 
 exit:
+    if (error != kErrorNone)
+    {
+        LogCrit("Failed to process Service Discovery instructions: %s", ErrorToString(error));
```
Comment (simonlingoogle) [inline src/core/net/srp_server.cpp:916]: ditto

File: src/core/net/srp_server.cpp [hunk 10]
```diff
@@ -989,6 +1012,11 @@ Error Server::ProcessServiceDescriptionInstructions(Host &           aHost,
     aMetadata.mOffset = offset;
 
 exit:
+    if (error != kErrorNone)
+    {
+        LogCrit("Failed to process Service Description instructions: %s", ErrorToString(error));
```
Comment (simonlingoogle) [inline src/core/net/srp_server.cpp:1017]: ditto

File: src/core/net/srp_server.cpp [hunk 11]
```diff
@@ -1067,6 +1095,11 @@ Error Server::ProcessAdditionalSection(Host *aHost, const Message &aMessage, Mes
     aMetadata.mOffset = offset;
 
 exit:
+    if (error != kErrorNone)
+    {
+        LogCrit("Failed to process DNS Additional section: %s", ErrorToString(error));
```
Comment (simonlingoogle) [inline src/core/net/srp_server.cpp:1100]: ditto

File: src/core/net/srp_server.cpp [hunk 12]
```diff
@@ -1113,6 +1146,11 @@ Error Server::VerifySignature(const Dns::Ecdsa256KeyRecord &aKeyRecord,
     error = aKeyRecord.GetKey().Verify(hash, signature);
 
 exit:
+    if (error != kErrorNone)
+    {
+        LogCrit("Failed to verify message signature: %s", ErrorToString(error));
```
Comment (simonlingoogle) [inline src/core/net/srp_server.cpp:1151]: ditto

File: src/core/net/srp_server.cpp [hunk 13]
```diff
@@ -1191,17 +1229,17 @@ void Server::SendResponse(const Dns::UpdateHeader &   aHeader,
...
     {
-        LogInfo("send success response");
+        LogInfo("Send success response");
     }
 
 exit:
     if (error != kErrorNone)
     {
-        LogWarn("failed to send response: %s", ErrorToString(error));
+        LogCrit("Failed to send response: %s", ErrorToString(error));
```
Comment (simonlingoogle) [inline src/core/net/srp_server.cpp:1242]: Ditto

**Review by simonlingoogle — APPROVED** [review]
LGTM 👍

File: src/core/net/srp_server.cpp (hunk 5 above)
Comment (abtink) [reply src/core/net/srp_server.cpp:329]: Yes. I agree. It should be full name (let's do it from a different PR).

Comment (abtink) [reply src/core/net/srp_server.cpp:329]: Digging a bit more, I realized we may be already doing the check correctly in the code 😃 

`Service::GetInstanceName()` returns the full service instance name and not just the "instance label". So we are checking for full name (e.g. `abc._serv1._udp` in  `host.HasServiceInstance(service.GetInstanceName()`).

```c++

        /**
         * This method gets the full service instance name of the service.
         *
         * @returns  A pointer service instance name (as a null-terminated C string).
         *
         */
        const char *GetInstanceName(void) const { return mDescription->mInstanceName.AsCString(); }
```

My suggestion is we try to harmonize our use of the terms "name" and "label" in the code (I think we may be using it in most places) 
- Basically "name" always refers to full name (not a single label)
- And use the term "label" when we refer to just one label (e.g. service instance label, or `abc` in `abc._serv1._udp`).

**Review by jwhui — APPROVED** [review]

File: src/core/net/srp_server.cpp (hunk 5 above)
Comment (abtink) [reply src/core/net/srp_server.cpp:329]: - Added PR https://github.com/openthread/openthread/pull/7588 to add this test (two clients registering services with same instance label).

Comment (abtink) [reply src/core/net/srp_server.cpp:329]: I also did not realize it for some time. I actually wrote a quick test-case, expecting it to fail, and was surprised why it was not failing  😄

Comment (simonlingoogle) [reply src/core/net/srp_server.cpp:329]: > I also did not realize it for some time. I actually wrote a quick test-case, expecting it to fail, and was surprised why it was not failing 😄

I guess now we have a real issue to fix.
